### DISCUSSION TITLE
Reserved Names with regex for Object.function()

### DISCUSSION THREAD

**Discussion by JonL1** [javascript-obfuscator/javascript-obfuscator#1156] [Q&A]
I'm trying to exclude all functions that belong to an object.

I'm using the following regex ```\.(.*)\(``` that should capture Object.**functionName**() to exclude them. However it doesn't work. Is it because this regex has two matching groups?  ```.functionName(``` and ```functionName```

Lookahead and lookbehind operators are off the table for compatibility reasons.

Any ideas?

**Comment by slig** [comment]
I'm not sure if the issue is the two matching groups, but I'd try (untested): `\.[^\(]+`.

**Reply by JonL1** [reply]
Thanks for the suggestion. Unfortunately that matches ```.functionName``` which doesn't work for reservedNames
